### PR TITLE
Add option to `.extract`: odict with pars when permuted=False

### PR DESCRIPTION
**Pull request by ahartikainen**
#### Summary:

Add option to extract specific parameters with `permuted=False`. 
This will return an odict with array shape = (n_save, chains, *(dims)).

Also fixed warning: `dtypes is None and permuted=False`--> `dtypes is not None and permuted=False and pars is None`

#### Intended Effect:

Extract specific pars keeping the chain order.
Makes setting up plots easier with huge number of parameters

#### How to Verify:

~~Tests~~ Tests added

#### Side Effects:

Silent bugs, where one uses `pars` with `permuted=False`

#### Documentation:

~~Needs updating~~ Updated

#### Reviewer Suggestions: 

@ariddell 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Ari Hartikainen

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: GPLv3 (http://opensource.org/licenses/GPL-3.0)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: pystan/stanfit4model.pyx [hunk 1]
```diff
@@ -551,19 +551,25 @@ cdef class StanFit4Model:
...
+        the sample array is for the iterations; the second for the number of chains;
+        the rest for the parameters. Parameters are listed in the same order as `pars`.
 
         """
         self._verify_has_samples()
         if inc_warmup is True and permuted is True:
             logging.warning("`inc_warmup` ignored when `permuted` is True.")
-        if dtypes is None and permuted is False:
-            logging.warning("`dtypes` ignored when `permuted` is False.")
+        if (dtypes is not None) and (permuted is False) and (pars is None):
```
Comment (ariddell) [inline pystan/stanfit4model.pyx]: I think this is fine: ``if dtypes is not None and not permuted and pars is None``. https://docs.python.org/3/reference/expressions.html#operator-precedence

**Comment by ariddell** [comment]
Looks good to me.

File: pystan/stanfit4model.pyx (hunk 1 above)
Comment (ahartikainen) [reply pystan/stanfit4model.pyx]: Good point, I will clean it

**Comment by ahartikainen** [comment]
@ariddel how can I restart appveyor?

**Comment by ariddell** [comment]
I restarted it. When you login to appveyor with github, do you not have permissions to restart it?

**Comment by ahartikainen** [comment]
In travis I have the option. Apparently not in Appveyor.

**Comment by ahartikainen** [comment]
Everything looks good. I will merge.

**Pull request merged**